buildSrc/src/main/groovy/org/gradle/build/UpdateReleasedVersions.groovy

```diff
@@ -21,24 +21,43 @@ import com.google.common.base.Preconditions
 import groovy.json.JsonBuilder
 import groovy.json.JsonSlurper
 import org.gradle.api.DefaultTask
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.gradle.util.GradleVersion
 
+import javax.inject.Inject
+
 class UpdateReleasedVersions extends DefaultTask {
 
     @Internal
     File releasedVersionsFile
 
     @Internal
-    ReleasedVersion currentReleasedVersion
+    Property<ReleasedVersion> currentReleasedVersion
+
+    @Inject
+    UpdateReleasedVersions(ObjectFactory objectFactory) {
+        currentReleasedVersion = objectFactory.property(ReleasedVersion)
```

Comment:
Should we use `project.objects.property` instead?

Reply:
From what I understood from @adammurdoch, we want to move more towards injecting domain objects instead of exposing more and more methods on `Project`.

The Javadocs list this way of consuming `ObjectFactory` first:

> An instance of the factory can be injected into a task or plugin by annotating a public constructor or method with javax.inject.Inject. It is also available via Project.getObjects().

Reply:
👍